### ISSUE TITLE
[BUG] Unexpected keyword argument 'dynamic' in labeled image parser

### ISSUE THREAD

**Issue by radao** [voxel51/fiftyone#2302]
### System information

-   **OS Platform and Distribution**: OSX 12.4
-   **Python version**: 3.10
-   **FiftyOne version** (`fiftyone --version`): 0.18.0
-   **FiftyOne installed from** (pip or source): pip

### Describe the problem

Using `dataset.add_labeled_images` results in an unexpected keyword argument 'dynamic'.

### Code to reproduce issue

```
import fiftyone as fo
import fiftyone.utils.data as foud

class LabeledImageSampleParser(foud.LabeledImageSampleParser):
    @property
    def has_image_path(self) -> bool:
        return True

    @property
    def has_image_metadata(self) -> bool:
        return False

    def get_image_path(self) -> str:
        return self.current_sample["image_path"]

    @property
    def label_cls(self) -> fo.Classification:
        return fo.Classification

    def get_label(self) -> fo.Classification:
        return fo.Classification(label="my-label")

dataset = fo.Dataset()
samples = [{"image_path": "~/Downloads/u1f624_u1f920.png"}]
sample_parser = LabeledImageSampleParser()
dataset.add_labeled_images(samples, sample_parser)
```

### Other info/logs

Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Please do not use screenshots for sharing text. Code snippets should be used instead when providing tracebacks, logs, etc.

### What areas of FiftyOne does this bug affect?

-   [ ] `App`: FiftyOne application issue
-   [x] `Core`: Core Python library issue
-   [ ] `Server`: FiftyOne server issue

### Willingness to contribute

The FiftyOne Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the FiftyOne codebase?

-   [ ] Yes. I can contribute a fix for this bug independently
-   [x ] Yes. I would be willing to contribute a fix for this bug with guidance from the FiftyOne community
-   [ ] No. I cannot contribute a bug fix at this time

**Comment by brimoor** [comment]
Good catch! This was a regression in v0.18. I patched this in https://github.com/voxel51/fiftyone/pull/2304 and added some tests to prevent this from happening in the future
